Improve avo status docs and add GitHub actions example (#229)
avohq/docs: pages/implementation/avo-in-the-ci.mdx

---
layout: 'guide'
title: 'Avo in CI/CD'

# abstract: 'Learn how to use Avo to create a tracking plan and generate high quality, actionable data.'

---

import Link from '../../components/Link'; 

The Avo CLI includes a command, `avo status` , which you can use ad-hoc or run in your CI to monitor that all expected analytics events are in fact implemented in your code.

## What is `avo status` ?

When you implement your analytics events with <Link passHref href="/implementation/avo-functions-overview"><a>Avo Functions</a></Link>, aka the type-safe analytics wrappers, you can run `avo status` to check where in your code base the analytics calls are made, or if they are missing.

## How does avo status work?

- `avo status` runs through your code base, searches for instances Avo Functions, and returns an error if there is no found instance of one or more of the analytics events that are marked as "Implement with Avo" for this code base.

> - `avo status` searches in the folder where it's invoked and in the children subfolders. It requires Avo to be initialized in that folder, i.e. there should be your `avo.json` file.

Reviewer: ```suggestion
- `avo status` searches in the folder where it's invoked and in the children subfolders. It requires Avo to be initialized in that folder, i.e. your `avo.json` file should be there.
```

> - It validates tracking calls implemented with Avo Function (the Avo generated code)

Reviewer: ```suggestion
- It validates tracking calls implemented with <Link passHref href="/implementation/avo-functions-overview"><a>Avo Functions</a></Link>, aka the Avo generated type-safe analytics wrappers.
```

└─ java android (musicplayerexample/src/main/java/sh/avo/Avo.java)
   ├─ appOpened
   │  └─ used in musicplayerexample/src/main/java/app/avo/musicplayerexample/MusicPlayerExampleApplication.kt: 1 time
   ├─ login
   │  └─ ✖ no usage found
   ├─ logout
   │  └─ ✖ no usage found
   ├─ pause
   │  └─ used in musicplayerexample/src/main/java/app/avo/musicplayerexample/ExampleMusicPlayerActivity.kt: 1 time
   ├─ play
   │  └─ used in musicplayerexample/src/main/java/app/avo/musicplayerexample/ExampleMusicPlayerActivity.kt: 3 times
   ├─ playNextTrack
   │  └─ used in musicplayerexample/src/main/java/app/avo/musicplayerexample/ExampleMusicPlayerActivity.kt: 1 time
   └─ playPreviousTrack
      └─ used in musicplayerexample/src/main/java/app/avo/musicplayerexample/ExampleMusicPlayerActivity.kt: 1 time
info 5 of 7 events seen in code
error 2 missing events
└─ java android (musicplayerexample/src/main/java/sh/avo/Avo.java)
   ├─ login: no usage found
   └─ logout: no usage found
```

<p>
  <Link passHref href="/help/troubleshooting"><a>Get in touch</a></Link>{' '}
  if you want to learn more about how Avo prevents analytics errors.
</p>

##  Integrating Avo with my CI

> > 💡 Why would I add Avo to my CI/CD? 

Reviewer: ```suggestion
> 💡 **Why would I add Avo to my CI/CD?**
```
